test/jdk/java/util/zip/ZipInputStream/ZipInputStreamCRCCheck.java

```diff
@@ -0,0 +1,158 @@
+
+            final ZipEntry entry = new ZipEntry(entryName);
+            entry.setSize(entrySize);
+            entry.setMethod(entryMethod);
+
+            final byte[] entryContent = new byte[entrySize];
+            final CRC32 crc = new CRC32();
+            crc.update(entryContent);
+            entry.setCrc(crc.getValue());
+
+            zos.putNextEntry(entry);
+            zos.write(entryContent);
+            zos.closeEntry();
+        }
+        final byte[] zipContent = baos.toByteArray();
+        // now change the CRC value of the entry to an incorrect value
+        changeEntryCRCValue(zipContent);
+        // use ZipInputStream to read the entry from that ZIP containing the incorrect CRC value
+        try (final ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(zipContent))) {
+            final ZipEntry entry = zis.getNextEntry();
+            assertEquals(entryName, entry.getName(), "unexpected entry name");
+            final ZipException zipEx = assertThrows(ZipException.class, () -> {
+                zis.readAllBytes();
+                zis.closeEntry();
+            }, "expected to fail due to CRC error but didn't, computed crc=" + entry.getCrc());
+            assertNotNull(zipEx.getMessage(), "missing exception message");
+            assertTrue(zipEx.getMessage().contains("invalid entry CRC"),
+                    "Expected exception message to contain 'invalid entry CRC' but was "
+                            + zipEx.getMessage());
+        }
+    }
+
+    /*
+     * Updates the CRC value of the (sole) entry in the ZIP to an incorrect value
+     */
+    private static void changeEntryCRCValue(final byte[] zipContent) {
+        final ByteBuffer bb = ByteBuffer.wrap(zipContent).order(ByteOrder.LITTLE_ENDIAN);
+        final boolean hasDataDescriptor = (bb.position(6).getShort() & 8) != 0;
+        // the offset corresponding to either the entry's LOC or the entry's data descriptor
+        // where the CRC value should be updated
+        final int crcOffset;
+        // the CRC value that's currently present at the determined CRC offset
+        final int currentCRC;
+        if (hasDataDescriptor) {
+            final int ddSigStart = findDataDescriptor(bb);
+            // CRC value resides 4 bytes after the start of the data descriptor signature
+            crcOffset = ddSigStart + 4;
+        } else {
+            // CRC value resides 14 bytes after the start of the LOC (which starts at the
+            // beginning of the ZIP content)
+            crcOffset = 14;
+        }
+        currentCRC = bb.position(crcOffset).getInt();
+        // update the CRC value to an arbitrary value
+        bb.position(crcOffset).putInt(currentCRC + 1);
+        System.out.println("tampered the crc value"
+                + " from 0x" + Integer.toHexString(currentCRC)
+                + " to 0x" + Integer.toHexString(bb.position(crcOffset).getInt()));
```

Comment:
```suggestion
        System.out.format("tampered the crc value from 0x%x to 0x%x%n", currentCRC, bb.position(crcOffset).getInt());
```
The test code does not require high performance, we can System.out.format